### PR TITLE
cherry picked partitioner

### PR DESCRIPTION
**Pull request by alan-stokes**
changes from roberts conversion for allowing a application vertex to partiton itself. includes support for multiple types of vertex with different projection mappings

tied to
PAC: https://github.com/SpiNNakerManchester/PACMAN/pull/260
FEC: https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/494
SPYNNAKER: https://github.com/SpiNNakerManchester/sPyNNaker/pull/708
SPYNNAKER8:

### CODE REVIEW

File: p8_integration_tests/quick_test/test_connectors/test_fixed_number_pre_connector.py [hunk 1]
```diff
@@ -68,85 +66,6 @@ def check_other_connect(self, connections, with_replacement):
                            allow_self_connections=True)
         sim.end()
 
-    def test_replace_self(self):
```
Comment (Christian-B) [inline p8_integration_tests/quick_test/test_connectors/test_fixed_number_pre_connector.py:71]: Why are these removed?

Comment (alan-stokes) [reply p8_integration_tests/quick_test/test_connectors/test_fixed_number_pre_connector.py]: resolved

**Review by andrewgait — APPROVED** [review]
These changes look fine to me.

**Review by dkfellows — APPROVED** [review]
